lib/set.rb

```diff
@@ -80,6 +83,12 @@ def self.[](*ary)
   #
   # If a block is given, the elements of enum are preprocessed by the
   # given block.
+  #
+  #     Set.new [1, 2]                        # -> #<Set: {1, 2}>
```

Comment:
Regarding style: there is not a strict convention throughout all docs, but what I observe for bigger parts of it:

``` ruby
5   #=> 5 (instead of `# ->`)
Set.new([1, 2])   # generally parentheses are used for method args
Set.new(1..10)
Set.new([1, 2, 3]) {|x| x * x }   # no space before |x|
[1, "c", :s]   # space between array elements
```

Also, double quotes are increasingly used for all strings, not only for interpolation.

Reply:
Hrm, I think `Set[1, 2, 3]` is idiomatic rather than `Set.new([1, 2, 3])`.

Reply:
The examples above are taken from the docs for `Set.new` (and the other examples use `Set.[]` in most cases).

Reply:
@stomar 

> #=> 5 (instead of `# ->`)

It is not done like that in the `Set` class..

> generally parentheses are used for method args

True, and it is also in the [Ruby style guide](https://github.com/bbatsov/ruby-style-guide#method-invocation-parens). I'll change it. 👍 

> Set.new([1, 2, 3]) {|x| x * x }   # no space before |x|

I do not know were you take this rule from, there are many examples with the parentheses in the Array class for example. And the [Ruby Style guide](https://github.com/bbatsov/ruby-style-guide#spaces-braces) says it should be written with spaces

> space between array elements

Opps! 🙈 I'll change it! 👍

Reply:
> It is not done like that in the Set class.

Yes, but there is only a _single_ block of examples in the Set class currently (before the PR), so this argument seems not too strong to me; `#=>` is nowadays used in the majority of the docs (similar to what `irb` outputs).

> I do not know were you take this rule from

... granted, I guess this depends on which other classes you look at, both versions are common.

Reply:
@stomar

> Yes, but there is only a single block of examples in the Set class currently (before the PR), so this argument seems not too strong to me; #=> is nowadays used in the majority of the docs (similar to what irb outputs).

ok, then I'll create a different commit to change this syntax. It is true that it is better to modify this now than later when there are many other example. 👍 

@stomar I really appreciate that you take the time to look at my code, but I not sure if it makes sense to discuss about style in a project that is not following style conventions. I will make the changes, but I would really like that Ruby documentation follows [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide) or equivalently [Rubocop rules](https://github.com/bbatsov/rubocop) as they are driven by the Ruby community and it would be consistent and helpful when developing that Ruby documentation follows it as well. This way we wouldn't need to discuss anything about style in PRs. I just open an issue for this: https://bugs.ruby-lang.org/issues/14112